pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -91,6 +91,11 @@ func (configgen *ConfigGeneratorImpl) BuildHTTPRoutes(
 			rc := configgen.buildGatewayHTTPRouteConfig(node, req.Push, routeName)
 			if rc != nil {
 				rc = envoyfilter.ApplyRouteConfigurationPatches(networking.EnvoyFilter_GATEWAY, node, efw, rc)
+				for _, vs := range rc.VirtualHosts {
```

Comment:
config should not be mutated after applying patches IMO.

Reply:
+1

Reply:
sure. 
How about removing the logic to set `maxStreamDuration` from the `setTimeout()`? 
cc @ramaraochavali

Reply:
I think it does not work because
- with out setting GrpcTimeoutHeaderMax user can not use grpc-timeout header which is a common case when timeout is not specified in VS
- If we just set GrpcTimeoutHeaderMax and not set MaxStreamDuration the timeouts are still not honoured. That is a bug in Envoy https://github.com/envoyproxy/envoy/pull/15585.

Any change here we need think through more carefully. 

Can you please explain the use case a bit more, why you are not setting timeout in VS but want to set it via Envoy filter?

Reply:
>Can you please explain the use case a bit more, why you are not setting timeout in VS but want to set it via Envoy filter?

Service Entries are created dynamically for external services and It is convenient to create retry/timeout configuration for these clusters through envoyfilter, otherwise will have to generate a VS for each hostname/ServiceEntry for timeout.

Reply:
If the `GrpcTimeoutHeaderMax` set for http1.1, then what's the effect?
Or does it only for grpc?

Reply:
ok, I made the changes before applying filter patches. PTAL!

Reply:
> If the GrpcTimeoutHeaderMax set for http1.1, then what's the effect?
Or does it only for grpc?

It only affects gRPC